C#/Botkit/Microsoft.BotKit/BotWorker.cs

```diff
@@ -104,17 +103,39 @@ public async Task<object> Reply(IBotkitMessage message, string response)
         /// </summary>
         /// <param name="id">ID of dialog</param>
         /// <param name="options">Object containing options to be passed into the dialog</param>
-        public async void BeginDialog(string id, object options)
+        public async Task BeginDialogAsync(string id, Dictionary<string, object> options)
```

Comment:
In Node the parameter options is nullable. Maybe check for it not being null before using it below?